### ISSUE TITLE
The final result is 0.7747

### ISSUE THREAD

**Issue by LfieLike** [youngkyunJang/SPQ#6]
I am running spq_main.py on RTX3090,and the final result is 0.7747,There is a big gap with the 0.793 in the paper.

**Comment by LfieLike** [comment]
python main_SPQ.py --gpu_id=0 --batch_size=256 --N_books=8 --N_words=16 --eval_epoch=100

**Comment by LfieLike** [comment]
tqdm                      4.62.3 
pytorch                   1.9.0 
python                    3.7.11 
kornia                    0.6.1
torchvision               0.10.0 
packaging                 21.3

**Comment by youngkyunJang** [comment]
Hello, How many epochs have you trained?
And I think there might be some difference in kornia version

**Comment by LfieLike** [comment]
Thanks for your reply, I used the default settings and trained 5000 epochs, I will downgrade kornia and try again

**Comment by LfieLike** [comment]
> Hello, How many epochs have you trained? And I think there might be some difference in kornia version

hello，I have downgraded the kornia version and trained 4500 epochs, but the results are still not good, the max mAP is 0.7728，I didn't change the code except for Retrival.py, line 17。

**Comment by youngkyunJang** [comment]
Hello, I empirically find that about 3,000 epoch, the performance converges. Could you eval more often by setting eval_epoch less than 100? In my experience, on average, after 2,500 epoch, for 32-bit experiments, the mAP achieves at least 0.780

**Comment by youngkyunJang** [comment]
There may be errors, so I'll check the code just in case.

**Comment by youngkyunJang** [comment]
Could you change Kg.ColorJitter(brightness=0.4, contrast=0.4, saturation=0.4, hue=0.1, p=0.8), with Kg.ColorJitter(brightness=0.4, contrast=0.8, saturation=0.8, hue=0.2, p=0.8), and conduct experiments? I usually change the setup proposed in SimCLR with only change the brightness as half (0.8 -> 0.4) and in this code, I diminish others as half too.

**Comment by LfieLike** [comment]
> Could you change Kg.ColorJitter(brightness=0.4, contrast=0.4, saturation=0.4, hue=0.1, p=0.8), with Kg.ColorJitter(brightness=0.4, contrast=0.8, saturation=0.8, hue=0.2, p=0.8), and conduct experiments? I usually change the setup proposed in SimCLR with only change the brightness as half (0.8 -> 0.4) and in this code, I diminish others as half too.

Thanks again for your reply, I will try again as your suggestion

**Comment by LfieLike** [comment]
> Could you change Kg.ColorJitter(brightness=0.4, contrast=0.4, saturation=0.4, hue=0.1, p=0.8), with Kg.ColorJitter(brightness=0.4, contrast=0.8, saturation=0.8, hue=0.2, p=0.8), and conduct experiments? I usually change the setup proposed in SimCLR with only change the brightness as half (0.8 -> 0.4) and in this code, I diminish others as half too.

after 4600 epochs,the max mAP is 0.7623,I also ran the code on the V100 and got the same result.

**Comment by youngkyunJang** [comment]
Hmm.. It's wired. I'll check it and share the result to you within a week.

**Comment by LfieLike** [comment]
hello，Did you find the code error?

**Comment by youngkyunJang** [comment]
Yes I just updated the code in main_SPQ.py, I change the initial value of codewords and fc layers, and loss functions from Torch.functional to Torch.nn as my original implementation!

**Comment by LfieLike** [comment]
> Yes I just updated the code in main_SPQ.py, I change the initial value of codewords and fc layers, and loss functions from Torch.functional to Torch.nn as my original implementation!

Haha, thanks for solving my query, I will close this issues

**Comment by xiaosu-zhu** [comment]
Hi @LfieLike @youngkyunJang , I've got similar results on CIFAR-10. Do you guys have any suggestions? I have `mAP: 0.7673` with `2800` epochs. @LfieLike have you run to a higher result? Thanks. Also provide my package infos for you:

```
PyTorch: 1.10.1
torchvision: 0.11.2
kornia: 0.6.2
```

Should I downgrade to lower versions?

**Comment by LfieLike** [comment]
> Hi @LfieLike @youngkyunJang , I've got similar results on CIFAR-10. Do you guys have any suggestions? I have `mAP: 0.7673` with `2800` epochs. @LfieLike have you run to a higher result? Thanks. Also provide my package infos for you:
> 
> ```
> PyTorch: 1.10.1
> torchvision: 0.11.2
> kornia: 0.6.2
> ```
> 
> Should I downgrade to lower versions?

After I close the issue，I ran the code once again，and got mAP：0.773，I could not reproduce this work,so  I give up following this work。

**Comment by LfieLike** [comment]
@xiaosu-zhu

**Comment by xiaosu-zhu** [comment]
> @xiaosu-zhu

Thanks @LfieLike for your quick reply 👍 . I decide to switch to exactly their `pytorch/torchvision/kornia` version to see if it helps.

**Comment by yayaYsmile** [comment]
@xiaosu-zhu 
Thank you for your reply. My problem is that the output mAP and MAX mAP are 0 when using CIFAR10 data set

I don't know what went wrong.Have you ever had the same problem?
